### PR TITLE
new MooshroomData class

### PR DESCRIPTION
**Pull request by ShaneBeee**
### Description
Add MooshroomData for mooshroom variations

---
**Target Minecraft Versions:** 1.14+
**Requirements:** 1.14+
**Related Issues:** #2537

### CODE REVIEW

File: src/main/java/ch/njol/skript/entity/MooshroomData.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+import org.bukkit.entity.MushroomCow;
+import org.bukkit.entity.MushroomCow.Variant;
+import org.eclipse.jdt.annotation.Nullable;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.lang.Literal;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+
+public class MooshroomData extends EntityData<MushroomCow>{
```
Comment (Matocolotoe) [inline src/main/java/ch/njol/skript/entity/MooshroomData.java]:
```suggestion
public class MooshroomData extends EntityData<MushroomCow> {
```

Comment (ShaneBeee) [reply src/main/java/ch/njol/skript/entity/MooshroomData.java]: Good catch, fixed!

File: src/test/skript/tests/regressions/2538-new MooshroomData class.sk [hunk 2]
```diff
@@ -0,0 +1,5 @@
+test "entity data mooshroom":
```
Comment (bensku) [inline src/test/skript/tests/regressions/2538-new MooshroomData class.sk]: Missed in review. Refer to issue, not PR in regtest name.

**Comment by bensku** [comment]
Code looks ok, but the test you included does not pass on 1.9 and 1.12.

**Comment by ShaneBeee** [comment]
> Code looks ok, but the test you included does not pass on 1.9 and 1.12.

I might remove the test from this one, Im not exactly sure how I would make this work in 1.12.2 and below, since the brown mooshroom was added in 1.14.

**Comment by bensku** [comment]
After commenting here, I added a way to gate test by MC version. Not ideal, because it can't be 1.14+ or like, but that can be added later.

File: src/test/skript/tests/regressions/2538-new MooshroomData class.sk (hunk 2 above)
Comment (ShaneBeee) [reply src/test/skript/tests/regressions/2538-new MooshroomData class.sk]: just to verify, are you meaning the script name "2538-new MooshroomData class.sk" should refer to the issue, or the test name `test "entity data mooshroom":` or both?

Comment (ShaneBeee) [reply src/test/skript/tests/regressions/2538-new MooshroomData class.sk]: renamed!

**Pull request merged**